app/uk/gov/hmrc/agentclientrelationships/controllers/Relationships.scala

```diff
@@ -74,7 +77,7 @@ class Relationships @Inject()(service: RelationshipsService) extends BaseControl
     }
   }
 
-  def create(arn: Arn, mtdItId: MtdItId): Action[AnyContent] = Action.async { implicit request =>
+  def create(arn: Arn, mtdItId: MtdItId): Action[AnyContent] = agentOrClient.async { implicit request =>
```

Comment:
We need to check that the arn or mtditid supplied to the method match the arn or mtditid obtained from the session.

So if the user is an agent the 2 arms must be the same, else if a client the 2 client ids must match.

See agent-client-authorisation AgencyInvitationsController.forThisAgency() for an example arn check.